android/src/firebase/firestore/TitaniumFirebaseFirestoreModule.kt

```diff
@@ -91,6 +90,47 @@ class TitaniumFirebaseFirestoreModule : KrollModule() {
         }
     }
 
+    private fun addListener(params: KrollDict) {
+        val collection = params["collection"] as String
+        val document = TiConvert.toString(params["document"], "")
+        val subcollection = TiConvert.toString(params["subcollection"], "")
+
+        var fireCollection = Firebase.firestore.collection(collection)
+        if (subcollection != "" && document != "") {
+            fireCollection = Firebase.firestore.collection(collection).document(document)
+                .collection(subcollection)
+        }
+
+        fireCollection.addSnapshotListener { value, e ->
+            if (e != null) {
+                return@addSnapshotListener
+            }
+
+            if (value != null) {
+
+                val kd = KrollDict()
+                val kdItems = arrayOfNulls<Any>(value.documentChanges.size)
+                var i:Int = 0
+
+                value.documentChanges.forEach { item ->
+                    val d = KrollDict()
+                    val doc = KrollDict()
+                    item.document.data.toMap().forEach {
+                        d[it.key] = it.value
+                    }
+                    doc["name"] = item.document.id
```

Comment:
Should this be `name` ?

Reply:
`document` is better as this is the same document ID you add in eg. `getDocument()`. I've updated the PR